### PR TITLE
Refactoring zkfold-cardano

### PR DESCRIPTION
**Pull request by zlonast**
- [x] Delete assets and test-data
- [x] Update README
- [x] Write backend to plonk
- [x] Write backend to symbolic
- [ ] Update benchs
- [x] Update packages
- [x] Update e2e-test cli scripts
- [x] Update e2e-test docs
- [x] Add local-testnet
- [x] Update circuit examples
- [x] Update plutus scripts
- [x] Update off-chain code
- [x] Update on-chain code
- [x] Update tests

### CODE REVIEW

File: src/ZkFold/Cardano/Plonk.hs [hunk 1]
```diff
@@ -157,7 +158,9 @@ instance NonInteractiveProof PlonkPlutus where
                 + v * (s1_xi
                 + v * s2_xi))))
                 + u * z_xi
-                ) `mul` g0
+                ) `mul` g0 -- bls12_381_G1_generator ?
```
Comment (vlasin) [inline src/ZkFold/Cardano/Plonk.hs:161]: Yes

File: src/ZkFold/Cardano/Plonk.hs [hunk 2]
```diff
@@ -157,7 +158,9 @@ instance NonInteractiveProof PlonkPlutus where
...
 
-            p1 = bls12_381_millerLoop (xi `mul` proof1 + (u * xi * omega) `mul` proof2 + f - e) h0
+            p1 = bls12_381_millerLoop (xi `mul` proof1 + (u * xi * omega) `mul` proof2 + f - e) h0 -- bls12_381_G2_generator ?
```
Comment (vlasin) [inline src/ZkFold/Cardano/Plonk.hs:163]: Also yes

**Comment by vlasin** [comment]
@zlonast, I added `CompatibleNonInteractiveProofs` to `zkfold-base`. You can use it for testing.

**Comment by vlasin** [comment]
Are we generating all assets in this repository? If so, maybe include them in `.gitignore`?

File: README.md [hunk 3]
```diff
@@ -37,6 +21,29 @@ $ cabal build
 
 ## To run
 
-1. `cabal run test`  to check the implementations against the test data.
-2. `cabal run bench-uplc`        to compile the fast implementation to UPLC (`.flat`) for further processing.
-3. `cabal run bench-cpu-mem`        to calculate the cpu/mem units of the tests in the plutus-benchmark package.
+## Import circom from zkfold-base
```
Comment (vlasin) [inline README.md:24]: Why circom?

File: backends/symbolic/symbolic-init-transaction/Main.hs [hunk 4]
```diff
@@ -48,14 +59,15 @@ import           ZkFold.Base.Algebra.EllipticCurve.BLS12_381 (BLS12_381_Scalar,
...
-import           ZkFold.Cardano.ScriptsVerifier              (RedeemerVerifier (..))
+import           ZkFold.Cardano.Plonk.OffChain               (Contract (..), PlonkN, RowContractJSON, mkInput, mkProof, mkSetup, toContract)
+import           ZkFold.Cardano.ScriptsVerifier              (RedeemerSymbolic (..))
 import           ZkFold.Symbolic.Cardano.Types               (TxId (..))
 import           ZkFold.Symbolic.Compiler                    (ArithmeticCircuit (..), compile)
 import           ZkFold.Symbolic.Compiler.ArithmeticCircuit  (applyArgs)
 import           ZkFold.Symbolic.Data.Bool                   (Bool (..))
 import           ZkFold.Symbolic.Data.Eq                     (Eq (..))
 import           ZkFold.Symbolic.Types                       (Symbolic)
+import ZkFold.Base.Data.Vector (Vector(..))
```
Comment (vlasin) [inline backends/symbolic/symbolic-init-transaction/Main.hs:70]: formating

File: README.md [hunk 5]
```diff
@@ -8,22 +8,6 @@ An integration of zkFold Symbolic smart contracts with the Cardano blockchain. T
 
 This project can be built with Cabal 3.10.2.1 and GHC 9.6.3.
```
Comment (vlasin) [inline README.md:9]: Please specify which components are currently working.

File: benchs/bench-cpu-mem/Main.hs [hunk 6]
```diff
@@ -18,14 +19,12 @@ import           ZkFold.Base.Protocol.ARK.Plonk              (Plonk (..), PlonkW
...
-import           ZkFold.Cardano.Plonk.OffChain               (Contract (..), Plonk32, RowContractJSON, mkInput, mkProof, mkSetup, toContract)
-import           ZkFold.Cardano.ScriptsVerifier              (DatumVerifier (..), RedeemerVerifier (..))
-import           ZkFold.Symbolic.Cardano.Types               (TxId (..))
-import           ZkFold.Symbolic.Compiler                    (ArithmeticCircuit (..), compile)
+import           ZkFold.Cardano.Plonk.OffChain               (Contract (..), PlonkN, RowContractJSON, mkInput, mkProof, mkSetup, toContract)
 import           ZkFold.Symbolic.Compiler.ArithmeticCircuit  (applyArgs)
 import           ZkFold.Symbolic.Data.Bool                   (Bool (..))
 import           ZkFold.Symbolic.Data.Eq                     (Eq (..))
 import           ZkFold.Symbolic.Types                       (Symbolic)
+import ZkFold.Base.Data.Vector (Vector(..))
```
Comment (vlasin) [inline benchs/bench-cpu-mem/Main.hs:27]: Formatting

File: local-testnet/scripts/babbage/alonzo-babbage-test-genesis.json [hunk 7]
```diff
@@ -0,0 +1,375 @@
+{
+  "collateralPercentage": 150,
+  "costModels": {
+      "PlutusV1": [
+          100788,
```
Comment (vlasin) [inline local-testnet/scripts/babbage/alonzo-babbage-test-genesis.json:5]: I believe the cost model parameters have names. Where did you take this file from?

File: README.md (hunk 3 above)
Comment (zlonast) [reply README.md:24]: oh it must be just a circuit...

File: local-testnet/scripts/babbage/alonzo-babbage-test-genesis.json (hunk 7 above)
Comment (zlonast) [reply local-testnet/scripts/babbage/alonzo-babbage-test-genesis.json:5]: https://github.com/IntersectMBO/cardano-node/blob/b16193878b118d851d2c9a8cf1ddd52be32697e0/scripts/babbage/conway-babbage-test-genesis.json#L4

Comment (zlonast) [reply local-testnet/scripts/babbage/alonzo-babbage-test-genesis.json:5]: > I believe the cost model parameters have names.

you're right, but for some reason the guys who wrote the parameters above didn't do it, although they could ¯\_(ツ)_/¯

**Pull request merged**